### DISCUSSION TITLE
How to determine if any form field is edited after submission if the form fields are filled with data? (v6)

### DISCUSSION THREAD

**Discussion by mg901** [react-hook-form/react-hook-form#2604] [Q&A] [answered]
I fill out the form and save it as a draft. After saving, the form is not cleared (data in the form is saved) next to me I show the block with documents, but as soon as the form starts to be edited, I hide the block of documents. How to determine if any form field is edited after submission if the form fields are filled with data?

**Comment by ehaynes99** [comment]
Good timing! I was just coming to ask something similar. A simpler example is that I have a submit button that is disabled while the form is invalid. On submit, I want to return it to disabled unless there is another edit.

In effect, I need a `isDirtySinceSubmit` flag, or some way to achieve the same. 

Thus far, all I have been able to come up with is a deep comparison of the last submitted value with the value of `watch()`, which is obviously not the most efficient.
```javascript
const canSubmit = formState.isValid && formState.isDirty && !isEqual(watch(), lastSubmitted)
```
https://codesandbox.io/s/crazy-bhaskara-58h72?file=/src/App.js

**Comment by mg901** [comment, marked as answer]
@ehaynes99 Thank you for good idea. I manage to solve our problem easer. [Just reset the data after submitting](https://codesandbox.io/s/kind-blackwell-bsdpc?file=/src/App.js).

**Reply by ehaynes99** [reply]
Oh, nice. "Resetting" to the same thing. ;)

This does reduce the render count as well, since `useForm` won't regenerate on every single keystroke, only when validation state changes.

**Reply by Shaker-Hamdi** [reply]
@mg901 The example link is borken.
Could you please post it again?

**Reply by Moshyfawn** [reply]
Should be something along the lines of

```js
const { formState } = useForm()
const { isSubmitSuccessful } = formState

useEffect(() => {
  if (isSubmitSuccessful) {
    reset(data)
  }
}, [data, isSubmitSuccessful])

```
